Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -18,276 +17,274 @@
+
+        demisto.debug(f'Setting new token in integration context, token valid until: {valid_until}')
         set_integration_context(integration_context)
-        return response.get("access_token")
-
-    def get_events(self, time_of_last_event_str, current_time):
-        token = self.get_token()
-        time_param = f'dg_time:{time_of_last_event_str},{current_time}'
-        headers = {
-            'Authorization': 'Bearer ' + token
-        }
-        response = self._http_request(
-            method='GET',
-            headers=headers,
-            params={'q': time_param}
+
+        return response['access_token']
+
+    def export_events(self, export_profile: str) -> dict:
+        """
+        Exports events for the export profile that arrived after setting the internal bookmark.
+
+        Args:
+            export_profile (str): Export profile name.
+
+        Returns:
+            dict: Response JSON.
+        """
+        return self._http_request(
+            method='POST',
+            url_suffix=f'/rest/2.0/export_profiles/{export_profile}/export',
+            raise_on_status=True,
         )
-        return response
 
+    def set_export_bookmark(self, export_profile: str) -> str:
+        """
+        Advances the internal bookmark / pointer for the export profile based on the last export events.
+        This is effectively an API-side implementation of `demisto.getLastRun` and `demisto.setLastRun`.
+
+        *Only call after successfully exporting events and sending them to XSIAM!*
 
-def test_module(client: Client, params: Dict[str, Any]) -> str:
+        Returns:
+            str: Response text.
+        """
+        return self._http_request(
+            method='POST',
+            url_suffix=f'/rest/2.0/export_profiles/{export_profile}/acknowledge',
+            raise_on_status=True,
+            resp_type='text',
+        )
+
+
+def test_module(client: Client, export_profiles: list[str]) -> str:
     """
-    Tests API connectivity and authentication'
-    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
-    successful.
-    Raises exceptions if something goes wrong.
+    Tests API connectivity and authentication.
     Args:
         client (Client): Digital Guardian client to use.
-        params (Dict): Integration parameters.
+        export_profiles (list): List of export profile names.
     Returns:
-        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+        str: 'ok' connection to the service is successful.
+    Raises:
+        DemistoException | HTTPError: If request failed.
     """
     try:
-        limit = arg_to_number(params.get('number_of_events')) or 1000
-        fetch_events(
-            client=client,
-            last_run={},
-            limit=limit,
-        )
+        for export_profile in export_profiles:
+            client.export_events(export_profile)
```

Comment:
Consider a try/except block for each profile, collecting the boolean test results into a list, logging faulty ones, and then return 'ok' if `all(results)` and raising exception otherwise. 

Any other way to list the valid/invalid profiles will work too

Reply:
Improved handling by creating a set of invalid profiles and raising a `DemistoException` if there are any.
See commit 34998072bf2cd42577c899aa6f60313b6ad86751